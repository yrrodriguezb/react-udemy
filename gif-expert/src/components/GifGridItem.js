import React from 'react'

export const GifGridItem = ({ id, title, url}) => {
  return (
    <div id={ id } className="card animate__animated animate__fadeIn">
       <img src={ url } alt={ title } />
       <p>{ title }</p>
    </div>
  )
}
