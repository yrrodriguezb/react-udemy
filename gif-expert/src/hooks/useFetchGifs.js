import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  })

  useEffect(() => {
	 	getGifs(category).then(gifs => {
       setTimeout(() => {
        setState({
          data: gifs,
          loading: false
        })
       }, 3000);
     });
	}, [ category ])

  return state;
}