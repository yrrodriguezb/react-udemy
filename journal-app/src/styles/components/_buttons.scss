.btn {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
      touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  background-image: none;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #fff;
  background-color: transparent;
  cursor: pointer;

  transition: color .3s ease;

  &:focus {
    outline: none;
  }

  &:hover {
    color: darken($color: white, $amount: 10)
  }
}

.btn-primary {
  background-color: $primary;
  border-radius: 3px;

  &:disabled {
    background-color: lighten($color: $primary, $amount: 15);
  }

  &:hover {
    background-color: darken($color: $primary, $amount: 15);
  }
}


.btn-block {
  width: 100%;
}

.google-btn {
  cursor: pointer;
  margin-top: 5px;
  width: 100%;
  height: 42px;
  background-color: $google-blue;
  border-radius: 2px;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);

  transition: box-shadow .3s ease;

  .google-icon-wrapper {
    position: absolute;
    margin-top: 1px;
    margin-left: 1px;
    width: 40px;
    height: 40px;
    border-radius: 2px;
    background-color: $white;
  }
  .google-icon {
    position: absolute;
    margin-top: 11px;
    margin-left: 11px;
    width: 18px;
    height: 18px;
  }
  .btn-text {
    float: right;
    margin: 11px 40px 0 0;
    color: $white;
    font-size: 14px;
    letter-spacing: 0.2px;
  }
  &:hover {
    box-shadow: 0 0 6px $google-blue;
  }
  &:active {
    background: $button-active-blue;
  }
}
