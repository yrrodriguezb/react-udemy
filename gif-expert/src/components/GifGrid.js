import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

	const { data:images, loading } = useFetchGifs(category);

  return (
		<React.Fragment>
			<h3 className='animate__animated animate__fadeIn'>{ category }</h3>

			{ loading && <a className='animate__animated animate__flash'>Loading</a> }

			{ 
				<div className='card-grid'>
					{
						images.map(img => (
							<GifGridItem 
								key={ img.id } 
								{ ...img } 
							/>
						))
					}
				</div> 
			}
		</React.Fragment>
  )
}
