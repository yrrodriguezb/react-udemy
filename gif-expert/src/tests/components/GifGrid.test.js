import React from 'react'
import "@testing-library/jest-dom"; // para el intellisense, por defecto ya viene importada
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';

// Simula la informacion que el useFetchGifs debe devolver
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en <GifGrid />', () => {
  const category = 'One Punch';
  
  test('debe mostrar el componente correactamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })


    const wrapper = shallow(<GifGrid category={ category } /> )
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
    const gifs = [
      { id: 1, title:'Test 1', url:'https://test1.com' },
      { id: 2, title:'Test 2', url:'https://test2.com' }
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    
    const wrapper = shallow(<GifGrid category={ category } /> );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })
})