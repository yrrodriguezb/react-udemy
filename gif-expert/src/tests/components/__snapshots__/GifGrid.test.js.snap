// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar el componente correactamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <a
    className="animate__animated animate__flash"
  >
    Loading
  </a>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Test 1"
      url="https://test1.com"
    />
    <GifGridItem
      id={2}
      key="2"
      title="Test 2"
      url="https://test2.com"
    />
  </div>
</Fragment>
`;
